samtranslator/model/__init__.py

```diff
@@ -536,6 +536,25 @@ def _construct_tag_list(
         # customer's knowledge.
         return get_tag_list(sam_tag) + get_tag_list(additional_tags) + get_tag_list(tags)
 
+    def propagate_tags(
+        self, propagate_tags: Optional[bool], resources: List[Resource], tags: Optional[Dict[str, Any]]
+    ) -> None:
+        """
+        Propagates tags to the resources.
+
+        :param resources: List of resources
+        :param tags: dictionary of tags to propagate to the resources.
+
+        :return: None
+        """
+        if not propagate_tags:
+            return
+
+        tags_list = get_tag_list(tags)
+        for resource in resources:
+            if "Tags" in resource.property_types:
+                resource.Tags = tags_list
```

Comment:
This assumes (1) the property is always called `Tags`, and (2) the property format is always the same.

Neither are true, for example:

1. In `AWS::Backup::Framework` it's called `FrameworkTags`: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-backup-framework.html#cfn-backup-framework-frameworktags
2. `AWS::AmplifyUIBuilder::Component` uses map for `Tags`: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-amplifyuibuilder-component.html#cfn-amplifyuibuilder-component-tags

For this reason I'd rather at the very least be abundantly clear on the assumptions, or somehow make it explicit.

Reply:
Will add comments to clearly state the assumptions I made here.

Reply:
Are you sure comment is enough?

I think the current PR will fail deployment; `AWS::ApiGatewayV2::Api` takes a map for `Tags`: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-api.html#cfn-apigatewayv2-api-tags

Reply:
I suggest we add a separate method for that assignment like
```python
@staticmethod
def assign_tags(reciever_resource, tags_list):
    if "Tags" in resource.property_types:
        resource.Tags = tags_list
```
So if we need to overload it to handle specific format for some sub-resources, we overload only it and don't copy-paste other code which is general

Reply:
I like it